e2e_samples/config_driven_data_pipelines/README.md

```diff
@@ -1,106 +1,188 @@
+# df = park.sql(f"select * from <table name>)
+df.show()
+
+
+# to print the tables created in processed zone
+schema_name = "customer_2_pz_fruit_app"
+
+tables = spark.sql(f"show tables in {schema_name}")
+
+tables_collection = tables.collect()
+for table in tables_collection:
+    table_name = table['tableName']
+    print(f"{schema_name}.{table_name}")
+
+# Uncomment and replace the <> with the table name printed from above and query on the table contents 
+# df = park.sql(f"select * from <table name>)
+df.show()
+
+# shutdown the notebook spark session
+spark.sparkContext.stop()
+spark.stop()
+```
+The sample table is like below: 
+ <img src= "./docs/images/Sample table output.png" width="2000px"/>
 
+## Known Issues, Limitations and Workarounds
 
-### Build wheel solution packages (*nix)
+### 1. Problem when developing with *cddp_solution* local package
+When import local developed module in your Python code, you'll normally pick one of below statements, both of them may introduce module-not-found errors if you run your code in wrong path, saying we run above batch job in other path than /src.
+```python
+from .utils.module_helper import find_class
+```
+```python
+from cddp_solution.common.utils.module_helper import find_class
+``` 
 
+To resolve above issues, we could install the local developed modules in editable mode.
+- Prepare *pyproject.toml* and *setup.cfg* files to define which Python files should be included in wheel package.
+- Run below command in path with the pyproject.toml file, to install local developed modules in editable mode.
 ```bash
-ls -ld src/cddp_solution tests/hon*/ | awk '{split($0, array); print array[9]}' | xargs -n 1 python -m build  -w
+pip install -e .
 ```
+Once it's installed properly, we could achieve below two advantages.
+- We could run/test source code in any path
+- Any new chanages in source code lines of the local installed modules could take effect directly with out run the pip-install command again, as long as the *setup.cfg* file itself keeps the same, otherwise we need to execute the above pip-install command again.
 
-
-### Running the API template Locally
-
-#### From IDE
-
-ToDo
-
-#### From Command Line
-
-```cmd
-pip install -r ./src/requirements.txt --user
-python ./src/setup.py build
-python ./src/init_app.py --host 0.0.0.0
+### 2. The Spark session running issue during local testing/debug
+If you are using the Devcontainer enviornment to run the sample locally, and if you refer to the test notebook above for local DB query, you might encounter the Spark running time issue and got errors like 
+```
+...
+Caused by: java.sql.SQLException: Failed to start database 'metastore_db' with class loader jdk.internal.loader.ClassLoaders$AppClassLoader@5ffd2b27, see the next exception for details.
+...
+```
+Remember to close the test notebook spark session to avoid the spark session conflct with the local dev environment. 

```

Comment:
Please add one sentence here.
"You can also clear the session by clicking the restart button of notebook kernel"

Reply:
Thanks a lot @siliang-j-1225 ! I have added the sentence with the diagram as illustration.